src/main/java/io/fabric8/maven/docker/WatchMojo.java

```diff
@@ -176,6 +176,9 @@ private Runnable createBuildWatchTask(final ServiceHub hub, final ImageWatcher w
             throws MojoExecutionException {
         final ImageConfiguration imageConfig = watcher.getImageConfiguration();
         final AssemblyFiles files = hub.getArchiveService().getAssemblyFiles(imageConfig, mojoParameters);
+        if (files.isEmpty()) {
+            log.error("No assembly files for " + imageConfig + ". Are you sure you invoked this goal after the `package` goal?");
+        }
```

Comment:
@jstrachan what do you think instead of adding a warning, to introduce to fork the Maven lifecycle for this goal so that package is automatically included and introduce a second `build-nofork` goal which can be used for binding in `<executions>` (where lifecycle forking is evil) ?

That the same way how e.g. the maven-source-plugin is designed.